### PR TITLE
Advanced Ecig Fix

### LINKED ISSUE #20830: Advanced electronic cigarette is much worse than its ordinary variant
First of all, advanced ecig doesn't cope with nicotine withdrawal at all.
Then, it doesn't have neither `fun` nor `stim` value. And at last, while its description states that it is still addictive, it doesn't have `addiction_type` and `addiction_potential`.

### PR DESCRIPTION
**Pull request by LetterShapedGlyphs**
#### Summary
```SUMMARY: Bugfixes "Implements functionality for existing Advanced Electronic Cigarette"```

#### Purpose of change
Fixes #20830 - Addiction mechanics are currently tied to comestibles and the Advanced Electronic Cigarette, as a tool, was unable to apply stimulant or addiction effects without a major rework.

#### Describe the solution
By creating a dummy ecig within iuse::ecig, the game is able to use the JSON from a regular Electronic Cigarette to trigger the appropriate consume_effects() without using a bunch of magic numbers or code duplication. Any changes to the behavior of a regular electronic cigarette should be automatically mirrored on the advanced one while preserving all functionality.

#### Describe alternatives you've considered
The ideal solution would be to embed the addiction effects into the nicotine_liquid directly and implement hooks for used TOOLS to trigger comestible style effects based on their AMMO. This would allow for variable concentrations in the nicotine_liquid as well as enable addiction for a few other stimulant delivery devices currently not being handled. As such this should be treated as a temporary fix until a more comprehensive solution can be implemented.

(Edited to fix the Summary line)

### CODE REVIEW

File: src/iuse.cpp [hunk 1]
```diff
@@ -433,6 +433,8 @@ int iuse::ecig( player *p, item *it, bool, const tripoint & )
             p->add_msg_if_player( m_neutral,
                                   _( "You inhale some vapor from your advanced electronic cigarette." ) );
             p->use_charges( "nicotine_liquid", 1 );
+            item dummy_ecig = item("ecig", int(calendar::turn));
+            p->consume_effects(dummy_ecig);
```
Comment (ZhilkinSerg) [inline src/iuse.cpp]:
```suggestion
            p->consume_effects( dummy_ecig );
```

File: src/iuse.cpp [hunk 2]
```diff
@@ -433,6 +433,8 @@ int iuse::ecig( player *p, item *it, bool, const tripoint & )
             p->add_msg_if_player( m_neutral,
                                   _( "You inhale some vapor from your advanced electronic cigarette." ) );
             p->use_charges( "nicotine_liquid", 1 );
+            item dummy_ecig = item("ecig", int(calendar::turn));
```
Comment (ZhilkinSerg) [inline src/iuse.cpp]:
```suggestion
            item dummy_ecig = item( "ecig", calendar::turn );
```

**Comment by AMurkin** [comment]
Suggested change:
```cpp
p->consume_effects( item( "ecig", it->birthday() ) );
```

File: src/iuse.cpp (hunk 2 above)
Comment (ZhilkinSerg) [reply src/iuse.cpp]: Cast to int is unnecessary - item constructor takes time_point.

File: src/iuse.cpp (hunk 1 above)
Comment (ZhilkinSerg) [reply src/iuse.cpp]: We use certain astyle rules.

File: src/iuse.cpp (hunk 2 above)
Comment (Karthas077) [reply src/iuse.cpp]: In my defense, with this one I literally copied that item creation line from the section immediately above.

Comment (ZhilkinSerg) [reply src/iuse.cpp]: Noone's attacking you :)

Existing code can (and will) be bad.

**Comment by Karthas077** [comment]
> Suggested change:
> 
> ```c++
> p->consume_effects( item( "ecig", it->birthday() ) );
> ```
Two things -
One: I feel like having it as a second line lends some additional clarity on what exactly the code is doing for if/when it gets changed later.

Two: I'd actually rather NOT use the birthday of the advanced electronic cigarette as the birthday of the dummy cigarette in the event that someone makes the basic ones expire (which, as a comestible, they could easily do and is something that would happen in real life). If anything, I'd prefer to use the birthday of the nicotine_liquid, but have no clue how to go about trying to get the birthday of a non-perishable liquid in a container in the player's inventory.

**Pull request merged**